### PR TITLE
sys/phydat: Fix unit confusion

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

Previously, `UNIT_G` was used for g-force with the correct symbol `g`, `UNIT_GR` for gram (as in kilogram) with the incorrect symbol `G` (which would be correct for Gauss), and `UNIT_GS` for Gauss with symbol `Gs` (which is an alternative correct symbol).

To avoid confusion between G-Force, Gauss, and Gram the units have been renamed to `UNIT_G_FORCE`, `UNIT_GRAM`, and `UNIT_GAUSS`. In addition, gram now uses the correct symbol `g`; which sadly is the same as for g-force. But usually there is enough context to tell them apart.

### Testing procedure

Green CI

### Issues/PRs references

None

### CODE REVIEW

**Review by benpicco — APPROVED** [review]

File: drivers/adxl345/adxl345_saul.c [hunk 1]
```diff
@@ -25,7 +25,7 @@ static int read_acc(const void *dev, phydat_t *res)
 {
     adxl345_read((const adxl345_t *)dev, (adxl345_data_t *)res->val);
 
-    res->unit = UNIT_G;
+    res->unit = UNIT_G_FORCE;
```
Comment (benpicco) [inline drivers/adxl345/adxl345_saul.c:28]: looks like you missed one in `tests/senml_phydat`

**Comment by benpicco** [comment]
doc in [`senml/phydat.h`](https://github.com/RIOT-OS/RIOT/pull/19292/files#diff-cc746a5597899ead782b13797658c5d12607d7f4759b73f7c011af96e86e3db3) needs an update

**Comment by benpicco** [comment]
`tests/phydat_dump` is the next one

**Review by kfessel — COMMENTED** [review]
https://en.wikipedia.org/wiki/Standard_gravity

ɡ0 or gn 

why rename gs to gaus ?

**Comment by benpicco** [comment]
IMO GAUSS and G_FORCE are much more readable than GS and GN

**Comment by maribu** [comment]
> ɡ0 or gn

Standard gravity and gravitational force equivalent is not the same thing. Using that would confuse more than help.

> why rename gs to gaus ?

Because the symbol for Gauss is G. So it should have been `UNIT_G`. But honestly, using the symbol in the macro name was a stupid idea to begin with. Is `UNIT_A` ampere, or is it Angstrom, or is it year (annus)? If we would have used `UNIT_AMPERE`, `UNIT_ANGSTROM`, and `UNIT_YEAR` instead, there would be no confusion.

**Comment by maribu** [comment]
bors merge

**Comment by kfessel** [comment]
g -> g0 is a bugfix

the rest might break someone else's compile, but do not fix a bug (therefor are api-change)

**Comment by kfessel** [comment]
@maribu: why did you mark this as a bug?

seems to me like phydat using short symbols might be some peoples feature

**Review by kfessel — COMMENTED** [review]
while changing you might also want to touch:

```
Check warning on line 15 in sys/senml/phydat.c  

too many consecutive empty lines
```

**Comment by benpicco** [comment]
Uh looks like the Rust riot-wrappers need an update too

**Comment by benpicco** [comment]
Having just gone though a lengthy bisect where those subtle API changes were the mayor pain point, I *really* want those compat defines.

**Review by kfessel — CHANGES_REQUESTED** [review]
please add compat defines

**Comment by maribu** [comment]
> Having just gone though a lengthy bisect where those subtle API changes were the mayor pain point, I _really_ want those compat defines.

I think here breaking existing users is quite beneficial, as they likely everyone using gram previously used `UNIT_G` before. Even if they would stumble upon the documentation that `UNIT_GR` is what they should have been using, they will directly revert that fix when this break phydat_str (outputing "G" instead of "g").

Breaking the code here gives users a chance to fix preexisting bugs.

**Comment by benpicco** [comment]
Hm do you have an indication that this was used wrong the whole time?
If *not* breaking compatibility comes with no hassle, I don't see why we should artificially burden users (if there are any - SAUL has always been a bit obscure and limited and most of those units are never used anywhere outside of tests, so this might well be a lot of spilled ink over nothing).

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
smpboot_control

**Comment by benpicco** [comment]
bors cancel

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
bors cancel

hmm looks like Rust doesn't think the compat define is enough :disappointed:

**Comment by kfessel** [comment]
I removed the minor tag since this change requires external users to change their code and the rust wrapper most likely also the rust users.

I think the "unit g-force" ist nonsens (physical and engineering but we may not care for these users in our phydat API)  (like kilowatts for energy  (instead of kWh) or amp-number for current)  g-force is a measurement in a specific context its unit is m/s²  the g is a a marker for the relation to g0 the same factor is used for shock or acceleration (which are not necessarily g-force) which ist most likely what the sensors are used for with riot.

I agree that the old api is not good (full name from the beginning like it was done for some of the units might have been better) but changing it for something that is also not good (even kind of wrong) while requiring a api change does not make sense to me.

I like to not care for this further and am sorry for the inconvenience I caused by discussing on this.

Most of this is not not a bugfix ( which is only the grams "G" to "g")

File: tests/unittests/tests-phydat/tests-phydat.c [hunk 2]
```diff
@@ -204,14 +204,14 @@ static void test_unitstr__success(void)
     TEST_ASSERT_EQUAL_STRING("m", phydat_unit_to_str_verbose(UNIT_M));
     TEST_ASSERT_EQUAL_STRING("m^2", phydat_unit_to_str_verbose(UNIT_M2));
     TEST_ASSERT_EQUAL_STRING("m^3", phydat_unit_to_str_verbose(UNIT_M3));
-    TEST_ASSERT_EQUAL_STRING("g", phydat_unit_to_str_verbose(UNIT_G));
```
Comment (chrysn) [inline tests/unittests/tests-phydat/tests-phydat.c:207]: The symbol Wikipedia, the [CGPM](https://www.bipm.org/en/committees/cg/cgpm/3-1901/resolution-2) and [The Brochure](https://www.bipm.org/documents/20126/41483022/SI-Brochure-9-EN.pdf/2d2b50bf-f2b4-9661-f402-5f9d66e4b507?version=1.11&t=1671101192839#page=45) use `g_n` (g subscript n) when not spelling it out.

Comment (benpicco) [reply tests/unittests/tests-phydat/tests-phydat.c:207]: We can do Unicode, so `gₙ`?

File: sys/include/phydat.h [hunk 3]
```diff
@@ -87,16 +87,16 @@ enum {
     UNIT_M2,        /**< square meters */
     UNIT_M3,        /**< cubic meters */
     /* kinetic */
-    UNIT_G,         /**< gravitational force */
+    UNIT_G_FORCE,   /**< gravitational force equivalent */
```
Comment (benpicco) [inline sys/include/phydat.h:90]:
```suggestion
    UNIT_G_FORCE,   /**< gravitational force equivalent */
    UNIT_G = UNIT_G_FORCE, /**< deprecated, use UNIT_G_FORCE instead */
```
This should hopefully work with c2rust too

File: sys/include/phydat.h [hunk 4]
```diff
@@ -87,16 +87,16 @@ enum {
...
     UNIT_DPS,       /**< degree per second */
     /* weight */
-    UNIT_GR,        /**< grams - not using the SI unit (kg) here to make scale
+    UNIT_GRAM,      /**< grams - not using the SI unit (kg) here to make scale
                      *   handling simpler */
```
Comment (benpicco) [inline sys/include/phydat.h:94]:
```suggestion
                     *   handling simpler */
    UNIT_GR = UNIT_GRAM, /**< deprecated, use UNIT_GRAM instead */
```

File: sys/include/phydat.h [hunk 5]
```diff
@@ -87,16 +87,16 @@ enum {
...
     /* electricity */
     UNIT_A,         /**< Ampere */
     UNIT_V,         /**< Volts */
     UNIT_W,         /**< Watt */
-    UNIT_GS,        /**< gauss */
+    UNIT_GAUSS,     /**< gauss */
```
Comment (benpicco) [inline sys/include/phydat.h:99]:
```suggestion
    UNIT_GAUSS,     /**< gauss */
    UNIT_GS = UNIT_GAUSS, /**< deprecated, use UNIT_GAUSS instead */
```

File: tests/unittests/tests-phydat/tests-phydat.c (hunk 2 above)
Comment (benpicco) [reply tests/unittests/tests-phydat/tests-phydat.c:207]: hmm looks like the Unicode is causing some trouble.
We already have m^3, so g_n would just be natural

Comment (chrysn) [reply tests/unittests/tests-phydat/tests-phydat.c:207]: Too bad ... any pointers to where the trouble manifested so we could fix things?

Comment (benpicco) [reply tests/unittests/tests-phydat/tests-phydat.c:207]: nvm just a [missed instance](https://github.com/RIOT-OS/RIOT/compare/229ece38e929bbdf14ac6372a1334acd1e55ddc5..90137d07e8def018c14be0abdad15a81591443e7)

File: sys/phydat/phydat_str.c [hunk 6]
```diff
@@ -106,14 +106,14 @@ const char *phydat_unit_to_str(uint8_t unit)
...
+        case UNIT_G_FORCE:  return "gₙ";
         case UNIT_DPS:      return "dps";
-        case UNIT_GR:       return "G";
+        case UNIT_GRAM:     return "g";
         case UNIT_A:        return "A";
         case UNIT_V:        return "V";
         case UNIT_W:        return "W";
         case UNIT_DBM:      return "dBm";
-        case UNIT_GS:       return "Gs";
+        case UNIT_GAUSS:    return "G";
```
Comment (kfessel) [inline sys/phydat/phydat_str.c:116]: Why would you go from less ambiguous to more?

```suggestion
        case UNIT_GAUSS:    return "Gs";
```

Comment (maribu) [reply sys/phydat/phydat_str.c:116]: `G` is the correct symbol for Gauss.

Comment (benpicco) [reply sys/phydat/phydat_str.c:116]: While both are correct, the actual SI unit for this is Tesla

Comment (maribu) [reply sys/phydat/phydat_str.c:116]: So deprecate `UNIT_GAUSS` right away?

Comment (benpicco) [reply sys/phydat/phydat_str.c:116]: No, and the conversion factor is 10000 anyway.
This PR started by fixing a *wrong* and making some units less ambiguous, it shouldn’t cause petty disruption.

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
bors merge

**Comment by miri64** [comment]
bors cancel
bors merge

**Comment by miri64** [comment]
bors cancel

**Comment by miri64** [comment]
bors merge

**Comment by maribu** [comment]
Thx :)

**Pull request merged**